test/unit/testUtil.js

```diff
@@ -52,11 +52,58 @@ define('testUtil', ['bluebird', 'json!/test/testConfig.json'], (Promise, TestCon
         return Promise.delay(timeMs);
     }
 
+    /**
+     * Wait for an element to appear in the DOM underneath `documentElement`
+     * @param {DOM element} documentElement to watch for the appearance of the element
+     * @param {string} selector to identify the element being watched for
+     * @returns {DOM element} the element
+     */
+    function waitForElement(documentElement, selector) {
+        return new Promise((resolve) => {
+            const element = document.querySelector(selector);
+            if (element) {
+                resolve(element);
+            }
+
+            const observer = new MutationObserver(() => {
```

Comment:
`[MutationObserver](https://developer.mozilla.org/en-US/docs/Web/API/MutationObserver)` tracks changes to the DOM. It keeps detailed records of what type of change has occurred, which can be queried; however, for these purposes, it is easier just to check whether the DOM element that we're interested in has appeared.